src/Common/OpenTelemetryTraceContext.h

```diff
@@ -23,16 +23,15 @@ struct Span
     UInt64 finish_time_us = 0;
     Map attributes;
 
-    void addAttribute(std::string_view name, UInt64 value);
-    void addAttributeIfNotZero(std::string_view name, UInt64 value);
-    void addAttribute(std::string_view name, std::string_view value);
-    void addAttributeIfNotEmpty(std::string_view name, std::string_view value);
-    void addAttribute(std::string_view name, std::function<String()> value_supplier);
-
-    /// Following two methods are declared as noexcept to make sure they're exception safe
-    /// This is because they're usually called in exception handler
-    void addAttribute(const Exception & e) noexcept;
-    void addAttribute(std::exception_ptr e) noexcept;
+    /// Following methods are declared as noexcept to make sure they're exception safe
+    /// This is because sometimes they will be called in exception handlers/dtor
+    bool addAttribute(std::string_view name, UInt64 value) noexcept;
+    bool addAttributeIfNotZero(std::string_view name, UInt64 value) noexcept;
+    bool addAttribute(std::string_view name, std::string_view value) noexcept;
```

Comment:
Add a `try` prefix to all of these methods? (cf. the discussion in issue #45308)

Reply:
Although `tryAddAttribute` is semantically perfect, changing the function name will introduce lots of changes. Let's keep the change in this PR small.